### PR TITLE
Support Solidity 0.8.18

### PR DESCRIPTION
**Pull request by taxio**
- [ ] Because this PR includes a **bug fix**, relevant tests have been included.
- [ ] Because this PR includes a **new feature**, the change was previously discussed on an Issue or with someone from the team.
- [x] I didn't do anything of this.

---

for https://github.com/NomicFoundation/hardhat/issues/3653

Bump up solidity version to [v0.8.18](https://github.com/ethereum/solidity/releases/tag/v0.8.18)

### CODE REVIEW

File: packages/hardhat-core/test/internal/hardhat-network/stack-traces/test.ts [hunk 1]
```diff
@@ -809,6 +809,11 @@ const solidityCompilers: SolidityCompiler[] = [
     compilerPath: "soljson-v0.8.17+commit.8df45f5f.js",
     latestSolcVersion: true,
   },
+  {
+    solidityVersion: "0.8.18",
+    compilerPath: "soljson-v0.8.18+commit.87f61d96.js",
+    latestSolcVersion: true,
```
Comment (alcuadrado) [inline packages/hardhat-core/test/internal/hardhat-network/stack-traces/test.ts:815]: This is the only thing that caught my attention.

**Review by alcuadrado — COMMENTED** [review]
Thanks for this PR! I only left a minor comment. Maybe @fvictorio can chime in.

File: packages/hardhat-core/test/internal/hardhat-network/stack-traces/test.ts (hunk 1 above)
Comment (fvictorio) [reply packages/hardhat-core/test/internal/hardhat-network/stack-traces/test.ts:815]: Not sure what you mean. The only thing that comes to mind about this is that maybe we should remove the `latestSolcVersion` flag from a previous version (e.g., 0.8.14).

Right now we don't have a policy about this. Maybe something like "the latest 3 versions of the current minor, and the last version of the previous minor" is enough.

Comment (fvictorio) [reply packages/hardhat-core/test/internal/hardhat-network/stack-traces/test.ts:815]: To clarify: this flag is used to filter the tested versions by default, but we also have a workflow that runs the stack traces tests with all the configured versions. So this is a sort of `fast` vs `full` tests config.

Comment (fvictorio) [reply packages/hardhat-core/test/internal/hardhat-network/stack-traces/test.ts:813]: I removed the `latestSolcVersion` flag from 0.8.14 and 0.8.15.

**Review by fvictorio — APPROVED** [review]

**Pull request merged**